tests/framework/di/ContainerTest.php

```diff
@@ -286,4 +287,26 @@ public function testContainerSingletons()
         $sameFoo = $container->get('qux.using.closure');
         $this->assertSame($foo, $sameFoo);
     }
+
+    /**
+     * @requires PHP 5.6
+     */
+    public function testVariadicConstructor()
+    {
+        $container = new Container();
+        $container->get(Variadic::class);
+    }
+
+    /**
+     * @requires PHP 5.6
+     */
+    public function testVariadicCallable()
+    {
+        $container = new Container();
+        $func = function(QuxInterface ...$quxes) {
```

Comment:
Whole tests fatal-errors because of it on PHP 5.4 and PHP 5.5. Any idea on how to test it safely?

Reply:
include the file only when PHP version is big enough. We did that for a yield test already: https://github.com/yiisoft/yii2/blob/0a7dbe3df5f40bb0e050ec01804645bfcccd4705/tests/framework/db/CommandTest.php#L307-L314